rules/scala_proto/private/ScalaProtoWorker.scala

```diff
@@ -50,11 +63,14 @@ object ScalaProtoWorker extends WorkerMain[Unit] {
       }
     }
 
-    ProtocBridge.runWithGenerators(
+    val exit = ProtocBridge.runWithGenerators(
       new MyProtocRunner,
       namedGenerators = List("scala" -> ScalaPbCodeGenerator),
       params = params
     )
+    if (exit != 0) {
```

Comment:
So this kills the entire worker now?

Reply:
No, because the base class registers an exit handler that catches and prevents the exit. It's weird...